### PR TITLE
popm: fix TestDisconnectedOpgeth

### PR DESCRIPTION
**Pull request by AL-CT**
**Summary**
Fixes the test `TestDisconnectedOpgeth`, as detailed in #718.

The test was run 100 times to ensure its viability: https://github.com/hemilabs/heminetwork/actions/runs/19041422367/job/54379408025

**Changes**
- Fixed a locking issue when closing websocket connections.
- Closing the connections did not necessarily prevent requests from being handled within a for loop, so added an exit condition when server is closed.
- Refactored the test to prevent other possible edge-cases and improve logging.

### CODE REVIEW

File: service/popm/popm_test.go [hunk 1]
```diff
@@ -351,22 +350,43 @@ func TestDisconnectedOpgeth(t *testing.T) {
...
+		case <-msgCh:
+		case rcvErr := <-errCh:
+			// miner has tried to reconnect
+			if errors.Is(rcvErr, mock.ErrConnectionClosed) {
+				t.Log("popminer attempted to reconnect")
+				opgeth.Start()
+				attempted = true
+			}
+		}
+	}
```
Comment (joshuasing) [inline service/popm/popm_test.go]: I think this can be simplified, maybe like below.
```suggestion
	var rcvErr error
	for {
		select {
		case <-ctx.Done():
			t.Fatal(ctx.Err())
		case <-msgCh:
			continue
		case rcvErr = <-errCh:
		}

		// miner has tried to reconnect
		if errors.Is(rcvErr, mock.ErrConnectionClosed) {
			t.Log("popminer attempted to reconnect")
			opgeth.Start()
			break
		}
	}
```

File: service/popm/popm_test.go [hunk 2]
```diff
@@ -351,22 +350,43 @@ func TestDisconnectedOpgeth(t *testing.T) {
...
+	for flushed := false; flushed != true; {
+		select {
+		case <-ctx.Done():
+			t.Fatal(ctx.Err())
+		case <-errCh:
+		default:
+			t.Log("error messages flushed")
+			flushed = true
+		}
 	}
```
Comment (joshuasing) [inline service/popm/popm_test.go]: I think this loop can be simplified, maybe like this:
```suggestion
	for {
		select {
		case <-ctx.Done():
			t.Fatal(ctx.Err())
		case <-errCh:
			continue
		default:
		}

		t.Log("error messages flushed")
		break
}
```

**Review by joshuasing — CHANGES_REQUESTED** [review]
